tyssue/solvers/viscous.py

```diff
@@ -118,9 +118,10 @@ def solve(self, tf, dt, on_topo_change=None, topo_change_args=()):
                 if "cell" in self.eptm.datasets:
                     self.history.record(["cell"], t)
                 self.eptm.topo_changed = False
-
             self.record(t)
 
+        self.history.closefile()
```

Comment:
instead of adding the method to the base history class, can't we test `if hasattr(history, "closefile"): ...` ?

Reply:
or maybe close in the record method? WIth a context manager?

Reply:
Actually, I didn't want to open and close the file at each recording time. But maybe it is better to do that, because in case of error the file can be never close.

Reply:
yes I think it's safer, and I'm not sure the overhead is very high...